packages/components/src/components/Header/Header.js

```diff
@@ -11,26 +11,65 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from 'react';
+import React, { Component } from 'react';
 
 import { Link } from 'react-router-dom';
-import { Header as CarbonHeader } from 'carbon-components-react';
-
+import { Button, Header as CarbonHeader } from 'carbon-components-react';
+import LogoutIcon from '@carbon/icons-react/lib/logout/20';
 import './Header.scss';
+import { shouldDisplayLogout } from '../../../../../src/api';
 
 import tektonBanner from '../../images/tekton-horizontal-color.png';
 
-export default function Header({ children }) {
-  return (
-    <CarbonHeader aria-label="Tekton Dashboard" className="header">
-      <Link to="/">
-        <img
-          alt="Tekton logo and title"
-          className="banner"
-          src={tektonBanner}
-        />
-      </Link>
-      {children && <div className="header-children">{children}</div>}
-    </CarbonHeader>
-  );
+export class Header extends Component {
+  state = {
+    showLogout: false
+  };
+
+  componentDidMount() {
+    this.determineOpenShift();
+  }
+
+  async determineOpenShift() {
+    try {
+      const detectedOpenShift = await shouldDisplayLogout();
+      this.setState({
+        showLogout: detectedOpenShift
+      });
+    } catch (error) {} // eslint-disable-line
+  }
+
+  handleLogoutClick() { // eslint-disable-line
+    window.location.href = '../../oauth/sign_out';
```

Comment:
Is this something we control? i.e. can we guarantee this relative path will be correct?

Reply:
I wonder if I can use base url/oauth/sign_out, checking